plugins/newsdownloader.koplugin/main.lua

```diff
@@ -1,16 +1,18 @@
-local WidgetContainer = require("ui/widget/container/widgetcontainer")
-local InfoMessage = require("ui/widget/infomessage")
-local UIManager = require("ui/uimanager")
 local DataStorage = require("datastorage")
 local FFIUtil = require("ffi/util")
-local util = require("util")
-local T = FFIUtil.template
-local _ = require("gettext")
-local logger = require("logger")
+local InfoMessage = require("ui/widget/infomessage")
+local LuaSettings = require("frontend/luasettings")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
 local ffi = require("ffi")
 local http = require("socket.http")
+local https = require("ssl.https")
+local logger = require("logger")
 local ltn12 = require("ltn12")
-local LuaSettings = require("frontend/luasettings")
+local socket_url = require("socket.url")
```

Comment:
I figured it was easier to use `socket_url` here than to change all of the already existing `url`s to something else.